syntax = "proto3";
package cadastrador;

service Cadastrador{
    rpc CadastraEscola (Escola) returns (DatabaseResponse) {};
    rpc CadastraAluno (Aluno) returns (DatabaseResponse) {};
    rpc ListaEscola (Void) returns (stream Escola) {};
    rpc ListaAluno (Void) returns (stream Escola) {};
    rpc RemoveEscola (ID) returns (DatabaseResponse) {};
    rpc RemoveAluno (ID) returns (DatabaseResponse) {};
}

message Void{

}

message ID{
    int32 id=1;
}

message DatabaseResponse{
    string message = 1;
}

message Aluno{
    ID escola = 1;
    string nome = 2;
    int32 gub = 3;
    string professor = 4;
}

message Escola{
    string nome = 1;
    string endereco = 2;
}